timemachine/fe/mcgregor.py

```diff
@@ -358,28 +360,33 @@ def mcs(
             filter_fxn,
         )
 
-        total_nodes_visited += mcs_result.nodes_visited
+        visits_left -= mcs_result.nodes_visited
 
-        # If timed out, either due to max_visits or max_cores, raise exception.
-        if mcs_result.timed_out:
-            warnings.warn(
-                f"Reached max number of visits/cores: {len(mcs_result.all_maps)} cores with {mcs_result.nodes_visited} nodes visited. "
-                "Cores may be suboptimal.",
-                MaxVisitsWarning,
-            )
+        total_nodes_visited += mcs_result.nodes_visited
 
         if len(mcs_result.all_maps) > 0:
+            # If we timed out but got cores, throw a warning
+            if mcs_result.timed_out:
+                warnings.warn(
+                    f"Reached max number of visits/cores: {len(mcs_result.all_maps)} cores with {mcs_result.nodes_visited} nodes visited. "
+                    "Cores may be suboptimal.",
+                    MaxVisitsWarning,
+                )
             # don't remove this comment and the one below, useful for debugging!
             # print(
             # f"==SUCCESS==[NODES VISITED {mcs_result.nodes_visited} | CORE_SIZE {len([x != UNMAPPED for x in mcs_result.all_maps[0]])} | NUM_CORES {len(mcs_result.all_maps)} | NUM_EDGES {mcs_result.num_edges} | time taken: {time.time()-start_time} | time out? {mcs_result.timed_out}]====="
             # )
             break
+        elif mcs_result.timed_out:
+            # If timed out, either due to max_visits or max_cores, raise exception.
+            raise NoMappingError(f"Exceeded max number of visits/cores: {mcs_result.nodes_visited} nodes visited.")
```

Comment:
I'd write this as `Exceeded max number of visits - no valid cores could be found: {mcs_result.nodes_visited} nodes visited.`